src/averageAges.js

```diff
@@ -37,7 +51,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothersNames = people
+    .filter(person => person.mother !== null)
+    .map(person => person.mother);
+
+  const women = people
+    .filter(person => person.sex === FEMALE)
+    .filter(person => withChildren ? mothersNames.includes(person.name) : true);
```

Comment:
The second time you filter only in case the withChildren parameter is passed, in another case you just iterate without any sense. So it's better to filter only if you actually need it.